fmriprep/utils/bids.py

```diff
@@ -68,14 +68,24 @@ def collect_derivatives(
             continue
         derivs_cache[f'{k}_boldref'] = item[0] if len(item) == 1 else item
 
+    transforms_cache = {}
     for xfm, q in spec['transforms'].items():
-        query = {**q, **entities}
-        if xfm == 'boldref2fmap':
-            query['to'] = fieldmap_id
-        item = layout.get(return_type='filename', **q)
+        # Transform extension will often not match provided entities
+        #   (e.g., ".nii.gz" vs ".txt").
+        # And transform suffixes will be "xfm",
+        #   whereas relevant src file will be "bold".
+        query = {
+            **q,
+            **{k: v for k, v in entities.items() if k not in ['suffix', 'extension']},
+        }
```

Comment:
I think we have this backwards, here and in L65, as we want the spec to provide overrides, not be overridden.

```suggestion
        query = {**entities, **q}
```

Reply:
Oh, lol, that is much better